### PR TITLE
feat: update cached resolver cache when not reading from the cache

### PR DESCRIPTION
**Pull request by ewanharris**
## Description

This PR introduces writing to the cache in `CachedCheckResolver` even when we are not reading from it due to the consistency requested being `HIGHER_CONSISTENCY`. Previously when we implemented skipping the cache we chose to not update the cache, but as it is relatively simple to do so we are now choosing to enrich the cache.

It refactors the existing test for this behaviour to now require the cache to be updated, and updates the list objects test that relied on cache interaction to reflect the new reality.

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

**Review by rhamzeh — APPROVED** [review]

File: internal/graph/cached_resolver.go [hunk 1]
```diff
@@ -150,37 +151,30 @@ func (c *CachedCheckResolver) ResolveCheck(
...
 		telemetry.TraceError(span, err)
 		return nil, err
 	}
 
-	cachedResp := c.cache.Get(cacheKey)
-	isCached := cachedResp != nil && !cachedResp.Expired()
-	span.SetAttributes(attribute.Bool("is_cached", isCached))
-	if isCached {
-		checkCacheHitCounter.Inc()
+	skipCache := c.enableConsistencyOptions && req.Consistency == openfgav1.ConsistencyPreference_HIGHER_CONSISTENCY
```
Comment (adriantam) [inline internal/graph/cached_resolver.go:161]: As an aside, we should have measurement tracking how frequently we have skip cache

File: internal/graph/cached_resolver.go [hunk 2]
```diff
@@ -6,13 +6,14 @@ import (
 	"strconv"
 	"time"
 
+	"go.opentelemetry.io/otel/attribute"
```
Comment (adriantam) [inline internal/graph/cached_resolver.go:9]: nitpik: why is this moved?

**Review by adriantam — APPROVED** [review]

File: internal/graph/cached_resolver.go (hunk 1 above)
Comment (ewanharris) [reply internal/graph/cached_resolver.go:161]: We have telemetry data that would allow us to count the total number of requests per consistency type (#1721), do you think we could use that to get the data you're thinking of?
